### PR TITLE
Add Golang job script printing Golang repo commit hash

### PR DESCRIPTION
**Pull request by tosi3k**
Small facilitation utility for debugging Golang-related regressions.

/sig scalability

### CODE REVIEW

File: golang/run-e2e.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+function print_golang_commit {
+  # Running "go version" on any of the extracted K8s binaries will give us
+  # the version of Go that's been used to build the binaries.
+  # In case of a non-standard, customly built Go, this will return the hash of
+  # the commit that was used for that Go build (plus some unnecessary junk that
+  # we remove below).
+  local -r go_version=$(go version $GOPATH/src/k8s.io/perf-tests/kubernetes/client/bin/kubectl)
+  local -r go_commit_hash=$(echo -n $go_version | awk '{ print $3 }' | cut -c2-)
+  local -r dashline=$(printf %100s | tr " " "-")
+  echo "${dashline}"
```
Comment (mm4tt) [inline golang/run-e2e.sh:30]: Can we also print the go_version? It wouldn't hurt, but will save us time if in the future the go version format changes and go_commit_hash will become gibberish

Comment (tosi3k) [reply golang/run-e2e.sh:30]: Good suggestion. Done.

**Pull request merged**